src/libpriv/rpmostree-core.c

```diff
@@ -878,6 +878,51 @@ checkout_pkg_metadata_by_nevra (RpmOstreeContext *self,
 }
 
 gboolean
+rpmostree_get_nevra_from_pkgcache (OstreeRepo  *repo,
+                                   const char  *nevra,
+                                   char       **out_name,
+                                   guint64     *out_epoch,
+                                   char       **out_version,
+                                   char       **out_release,
+                                   char       **out_arch,
+                                   GCancellable *cancellable,
+                                   GError  **error)
+{
+  g_autofree char *ref = NULL;
+  if (!find_cache_branch_by_nevra (repo, nevra, &ref, cancellable, error))
+    return FALSE;
+
+  g_autofree char *rev = NULL;
+  if (!ostree_repo_resolve_rev (repo, ref, FALSE, &rev, error))
+    return FALSE;
+
+  g_autoptr(GVariant) commit = NULL;
+  if (!ostree_repo_load_commit (repo, rev, &commit, NULL, error))
+    return FALSE;
+
+  g_autoptr(GVariant) meta = g_variant_get_child_value (commit, 0);
+  g_autoptr(GVariantDict) dict = g_variant_dict_new (meta);
+
+  g_autoptr(GVariant) nevra_variant =
+    g_variant_dict_lookup_value (dict, "rpmostree.nevra", G_VARIANT_TYPE ("a{sv}"));
```

Comment:
Why not just make it a variant of type `stsss`, i.e. a struct/tuple rather than a property bag?  We're never going to change what "nevra" means right?

Then you could just use `g_variant_get()` directly into the out params.